modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/StaticConfigurationInstanceProvider.java

```diff
@@ -29,6 +36,14 @@ public StaticConfigurationInstanceProvider(String name, Configuration configurat
         providerDelegate = new FirstTimeProviderDelegate<>();
     }
 
+    /**
+     * Returns {@link #configurationInstance}. The first time this method is invoked, the instance
```

Comment:
This is not clear to me. Where does the configuration instance come from? And from the description, the callback is only required on the first call